### ISSUE TITLE
linux: podman-remote silently fails to use RSA ssh keys

### ISSUE THREAD

**Issue by ssbarnea** [containers/podman#8499]
/kind bug

**Description**

**Steps to reproduce the issue:**

```
ssbarnea@leno:~
▶ CONTAINER_HOST=ssh://root@localhost/run/podman/podman.sock podman-remote --log-level=debug ps
INFO[0000] podman-remote filtering at log level debug
DEBU[0000] Called ps.PersistentPreRunE(podman-remote --log-level=debug ps)
DEBU[0000] Found SSH_AUTH_SOCK "/tmp/ssh-IBRTUifY3a/agent.1012473", ssh-agent signer enabled
Error: Failed to create sshClient: Connection to bastion host (ssh://root@localhost/run/podman/podman.sock) failed.: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain

# proof that ssh is able to connect correctly (current user has no local keys but is has agent)
ssbarnea@leno:~
▶ ssh root@localhost echo worked!
worked!

▶ ssh-add -l
4096 SHA256:IIgRCJ84QIlEIoYJ1RiFYRwPlbxXVr3z/7jo+FTM6zg /Users/ssbarnea/.ssh/id_rsa (RSA)
```

**Describe the results you received:**

Podman failed to establish ssh remote connection.

**Describe the results you expected:**

Podman should have suceeded establishing connection.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
▶ podman version
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15.2
Built:        Wed Oct  7 17:21:20 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
▶ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.21-3.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 0f53fb68333bdead5fe4dc5175703e22cf9882ab'
  cpus: 8
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: journald
  hostname: leno
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.16-300.fc33.x86_64
  linkmode: dynamic
  memFree: 19752214528
  memTotal: 33059995648
  ociRuntime:
    name: crun
    package: crun-0.15.1-1.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.15.1
      commit: eb0145e5ad4d8207e84a327248af76663d4e50dd
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-4.dev.giteecccdb.fc33.x86_64
    version: |-
      slirp4netns version 1.1.4+dev
      commit: eecccdb96f587b11d7764556ffacfeaffe4b6e11
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 20896014336
  swapTotal: 20896014336
  uptime: 722h 38m 27.91s (Approximately 30.08 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/ssbarnea/.config/containers/storage.conf
  containerStore:
    number: 13
    paused: 0
    running: 1
    stopped: 12
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.2.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
  graphRoot: /home/ssbarnea/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 63
  runRoot: /run/user/1000/containers
  volumePath: /home/ssbarnea/.local/share/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 1602087680
  BuiltTime: Wed Oct  7 17:21:20 2020
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 2.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.1.1-12.fc33.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by jwhonce** [comment]
@ssbarnea Are you running podman-remote on the same fedora 33 machine as the service?

`DEBU[0000] Found SSH_AUTH_SOCK "/tmp/ssh-IBRTUifY3a/agent.1012473", ssh-agent signer enabled`

should have gotten you in.

**Comment by ssbarnea** [comment]
While I already posted this on another macos related ticket, I think it is better suited here:

I suspect this is highly dependent on key type and size and apparently there is no debug option to enable to make it more verbose.

Confirmed that my both keys are 4096 rsa, one of them is encrypted at rest (loaded into agent) and the other one is unencrypted. podman refuses to connect to the bastion host with both, but ssh has no problems connecting.

The bastion is a fedora 33 machine but i doubt that matters much because ssh works, with RSA keys too!

Update
Now I do have the confirmation that podman is unable to establish ssh connection using RSA keys, regardless if they are loaded by itself or via ssh-agent, it fails to establish a handshake with them:

Error: Failed to create sshClient: Connection to bastion host (ssh://root@leno:22/var/run/podman/podman.sock) failed.: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
Following https://www.ssh.com/ssh/keygen/ I created two other keys ecdsa and ed25519 and they both worked. I skiped testing dsa as I mostly considered unsafe/not-recommended/deprecated.

Basically podman fails to use rsa keys, while ssh does work with them, regardless of what remote server you are trying to connect to. That seams like a bug in ssh implementation used by podman, especially as RSA is still the most commonly used key algorithm.

**Comment by philnalwalker** [comment]
I have confirmed the same behavior on my Macbook laptop where an ECDSA SSH key works, but RSA does not.

I was using podman-remote 2.1.1 installed via homebrew and a brand new Fedora Silverblue VM that comes default with Fedora Silverblue.

**Comment by jwhonce** [comment]
See https://github.com/containers/podman/issues/7806#issuecomment-742844281

Executive summary: --identity and ssh-agent work together, --log-level=debug now verbose about key handling to aid debugging, no RSA fix yet

**Comment by ssbarnea** [comment]
AFAIK this bug is still not sorted but it would be useful to see what is causing it. Clearly it has nothing to do with Fedora changer related to RSA because openssh can connect to Fedora boxes using RSA.

**Comment by rhatdan** [comment]
Any chance this has been fixed in podman 3.0?

**Comment by ssbarnea** [comment]
Apparently I had to renable the socker after upgrading to on fedora 33, as the socket was not longer open.  -- `systemctl enable --now podman.socket` 

But after doing this I tested with only one RSA key and it worked, so we can close it as fixed in 3.0 (or 3.0.1 for sure).

**Comment by dustymabe** [comment]
Honestly I'm thinking this has more to do with this upstream golang issue: https://github.com/golang/go/issues/37278
